libs/dal/src/repositories/base-repository.ts

```diff
@@ -9,7 +9,7 @@ import {
 import { Model, Types, ProjectionType, FilterQuery, UpdateQuery, QueryOptions } from 'mongoose';
 import { DalException } from '../shared';
 
-export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement = object> {
+export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
```

Comment:
removed the default value from the base, if any of the child classes are already on the env level they can pass empty object. this way we will make sure other classes are enforced to pass enforcement type